qadence_protocols/mitigations/readout.py

```diff
@@ -6,10 +6,35 @@
 import numpy.typing as npt
 from numpy.linalg import inv, matrix_rank, pinv
 from qadence import QuantumModel
+from qadence.logger import get_logger
 from qadence.noise.protocols import Noise
-from qadence.types import ReadOutOptimization
 from scipy.linalg import norm
 from scipy.optimize import LinearConstraint, minimize
+from scipy.sparse import csr_matrix
+from scipy.sparse.linalg import gmres
+
+from qadence_protocols.types import ReadOutOptimization
+
+logger = get_logger(__name__)
+
+
+def normalized_subspace_kron(noise_matrices: npt.NDArrary, subspace: npt.NDArray) -> npt.NDArray:
```

Comment:
```suggestion
def normalized_subspace(noise_matrices: npt.NDArrary, subspace: npt.NDArray) -> npt.NDArray:
```

Also: could you please add some docstring here ?